src/connectors/impls/gcl.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rand::Rng;
+use tremor_script::{tremor_fn, Registry, Value};
+
+pub(crate) mod writer;
+
+// The span ID within the trace associated with the log entry.
+//  For Trace spans, this is the same format that the Trace API v2
+//  uses: a 16-character hexadecimal encoding of an 8-byte array,
+// such as 000000000000004a
+fn random_span_id_value(ingest_ns_seed: u64) -> Value<'static> {
+    let mut rng = tremor_common::rand::make_prng(ingest_ns_seed);
+    let span_id: String = (0..8)
+        .map(|_| rng.gen_range(0_u8..=255_u8))
+        .map(|b| format!("{:02x}", b))
+        .collect();
+    Value::from(span_id)
+}
+
+fn random_trace_id_value(ingest_ns_seed: u64) -> Value<'static> {
```

Comment:
would it make sense to have a `random_id` function that both `random_trace_id_value` and `random_span_id_value` call

Reply:
Narrator: it wouldn't since they have a different size ...